src/views/templates/details/tabs/details/components/WorkloadProfile.tsx

```diff
@@ -0,0 +1,97 @@
+import * as React from 'react';
+import { ensurePath } from 'src/views/templates/utils';
+import { WorkloadProfileKeys } from 'src/views/templates/utils/constants';
+import { getWorkloadProfile } from 'src/views/templates/utils/selectors';
+
+import { TemplateModel } from '@kubevirt-ui/kubevirt-api/console';
+import VirtualMachineModel from '@kubevirt-ui/kubevirt-api/console/models/VirtualMachineModel';
+import { useModal } from '@kubevirt-utils/components/ModalProvider/ModalProvider';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { getTemplateVirtualMachineObject } from '@kubevirt-utils/resources/template';
+import { VM_WORKLOAD_ANNOTATION } from '@kubevirt-utils/resources/vm';
+import { k8sPatch } from '@openshift-console/dynamic-plugin-sdk';
+import {
+  Button,
+  DescriptionListDescription,
+  DescriptionListGroup,
+  DescriptionListTerm,
+} from '@patternfly/react-core';
+import { PencilAltIcon } from '@patternfly/react-icons';
+
+import { TemplateDetailsGridProps } from '../TemplateDetailsPage';
+
+import WorkloadProfileModal from './WorkloadProfileModal';
+
+const WorkloadProfile: React.FC<TemplateDetailsGridProps> = ({ template, editable }) => {
+  const { createModal } = useModal();
+  const { t } = useKubevirtTranslation();
+  const workload = getWorkloadProfile(template);
+
+  const updateWorkload = (updatedWorkload: WorkloadProfileKeys) => {
+    // const updatedTemplate = produce<V1Template>(template, (templateDraft: V1Template) => {
+    //   const vm = getTemplateVirtualMachineObject(templateDraft);
+    //   ensurePath(vm, 'spec.template.metadata.annotations');
+
+    //   vm.spec.template.metadata.annotations[VM_WORKLOAD_ANNOTATION] = updatedWorkload;
+
+    //   return templateDraft;
+    // });
+
+    // return k8sUpdate({
```

Comment:
@hstastna @yaacov Reading up on the docs, I think we could use `k8sUpdate` (replace a resource) when we have the full object and a potential optimistic lock failure (due to resource being modified between the read and write) is acceptable. If these conditions are true, it's easier to use `k8sUpdate` than `k8sPatch`.

Patching is more complex because we need to compute the patch meta object and IIRC there were some slight differences between regular resources vs. some specific sub resources. With multiple `k8sPatch` calls for the given resource and field combo, the last write will win. So if we're really concerned about potential optimistic lock failures and we don't have the full object, we should use `k8sPatch` method.

Reply:
in this case `k8sPatch` is straight forward to implement, IMHO it's worth the extra time to implement here.

for example the path can be:
```ts 
const workloadPath = `/objects/${vmObjectIndex}/spec/template/metadata/annotations/vm.kubevirt.io~1workload`;
```
and the patch comment can also be small:
```ts
 return k8sPatch({
      model: TemplateModel,
      resource: template,
      data: [
        {
          op: hasWorkload ? 'replace' : 'add',
          path: workloadPath,
          value: updatedWorkload,
        },
      ],
    });
```